app/assets/javascripts/account/AccountController.js.coffee

```diff
@@ -252,11 +253,22 @@ AccountController = (
     # which will auto-trigger its ui-validation
     day.$setViewValue(day.$viewValue + ' ')
 
+  $scope.isSale = (listing) ->
+    ListingIdentityService.isSale(listing)
+
+  $scope.hasSaleAndRentalApplications = (applications) ->
+    if _.first(applications)
+      firstIsSale = ListingIdentityService.isSale(_.first(applications).listing)
+      _.some applications, (application) ->
```

Comment:
so this is checking to see if any of the non-first applications are the opposite of the first? it's very clever but a bit confusing. I think it might benefit from a comment explaining it